File: src/algorithm/fcfs.js
export function fcfs(requests, head) {
    let totalMovement = 0;
    let sequence = [head]; // Start from initial head
  
    for (let i = 0; i < requests.length; i++) {
      totalMovement += Math.abs(requests[i] - head);
      head = requests[i];
      sequence.push(head);
    }
  
    return {
      totalMovement,
      sequence
    };
  }
  

File: backend/app/algorithms/fcfs.py
def fcfs(requests,head):
    sequence = [head]
    total_seek_time=0
    current=head 

    for request in requests:
        total_seek_time += abs(request - current)
        sequence.append(request)
        current = request
        average_seek_time = total_seek_time / len(requests)

    return {
        "sequence": sequence,
        "total_seek_time": total_seek_time,
        "average_seek_time": average_seek_time
    }